fiftyone/core/map/mapper.py

```diff
@@ -108,33 +122,51 @@ def map_samples(
             for sample in self._sample_collection.iter_samples(
                 progress=progress, autosave=save
             ):
-                result = map_fcn(sample)
-                yield sample.id, result
+                try:
+                    result = map_fcn(sample)
+                except Exception as err:
+                    if not skip_failures:
+                        raise err
+
+                    logger.warning(
```

Comment:
we should probably log the error before raising it, and maybe add worker id as well?

Reply:
This is an `or` situation. It either logs or raises. This is also sequentially mapping so there are no workers to log.